components/fxa-client/android/src/main/java/org/mozilla/fxaclient/internal/FirefoxAccount.kt

```diff
@@ -160,6 +160,122 @@ class FirefoxAccount : RustObject {
+    fun setDeviceDisplayName(displayName: String) {
+        rustCall { e ->
+            FxaClient.INSTANCE.fxa_set_display_name(validHandle(), displayName, e)
+        }
+    }
+
+    /**
+     * Retrieves the list of the connected devices in the current account, including the current one.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     */
+    fun getDevices(): Array<Device> {
+        val devicesBuffer = rustCall { e ->
+            FxaClient.INSTANCE.fxa_get_devices(validHandle(), e)
+        }
+        try {
+            val devices = MsgTypes.Devices.parseFrom(devicesBuffer.asCodedInputStream()!!)
+            return Device.fromCollectionMessage(devices)
+        } finally {
+            FxaClient.INSTANCE.fxa_bytebuffer_free(devicesBuffer)
+        }
+    }
+
+    /**
+     * Retrieves any pending commands for the current device.
+     * This should be called semi-regularly as the main method of commands delivery (push)
+     * can sometimes be unreliable on mobile devices.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @return A collection of [AccountEvent] that should be handled by the caller.
+     */
+    fun pollRemoteCommands(): Array<AccountEvent> {
+        val eventsBuffer = rustCall { e ->
+            FxaClient.INSTANCE.fxa_poll_remote_commands(validHandle(), e)
+        }
+        try {
+            val events = MsgTypes.AccountEvents.parseFrom(eventsBuffer.asCodedInputStream()!!)
+            return AccountEvent.fromCollectionMessage(events)
+        } finally {
+            FxaClient.INSTANCE.fxa_bytebuffer_free(eventsBuffer)
+        }
+    }
+
+    /**
+     * Handle any incoming push message payload coming from the Firefox Accounts
+     * servers.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @return A collection of [AccountEvent] that should be handled by the caller.
+     */
+    fun handlePushMessage(payload: String): Array<AccountEvent> {
+        val eventsBuffer = rustCall { e ->
+            FxaClient.INSTANCE.fxa_handle_push_message(validHandle(), payload, e)
+        }
+        try {
+            val events = MsgTypes.AccountEvents.parseFrom(eventsBuffer.asCodedInputStream()!!)
+            return AccountEvent.fromCollectionMessage(events)
+        } finally {
+            FxaClient.INSTANCE.fxa_bytebuffer_free(eventsBuffer)
+        }
+    }
+
+    /**
+     * Ensure the current device "Send Tab" commands has been registered with the server.
+     * This method should be called once per "device lifetime" after the Sync Keys have been
+     * obtained and called again if they change.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     */
+    fun ensureSendTabRegistered() {
+        rustCall { e ->
+            FxaClient.INSTANCE.fxa_ensure_send_tab_registered(validHandle(), e)
+        }
+    }
```

Comment:
FWIW, I haven't given up hope that we'll one day figure out how to factor this out into a separate module rather than growing a method off the main `FxAClient` object, perhaps after we get more experience with shipping multi-component megazord setups on different platforms.  But a big 👍 to getting it out the door this way to start with :-)